src/main/play-doc/operation/DynamicProxyConfiguration.md

```diff
@@ -0,0 +1,627 @@
+* *HAProxy Specific Configuration* section is populated with operating system specific setup. In the example above it is populated with a cut-down version of Ubuntu's configuration.
+* *Custom Endpoint Configuration* section contains application specific proxy configuration.
+* The endpoint that matches the following criteria has a custom configuration declared within the [ifAcl](#ifAcl) block:
+  * `system` is `public-site`
+  * `system version` is `1`
+  * `endpoint name` is `main`
+* The following takes place within custom configuration includes:
+  * HAProxy frontend `my_www_frontend` is configured to listen on HTTP port `80` and will relay request for `www.acme.com` to `my_www_backend`.
+  * HAProxy backend `my_www_backend` server is generated based on the backend server information contained within the [eachBackendServer](#eachBackendServer) helper function.
+  * The backend server information will be updated whenever there's changes to the endpoint, e.g. bundle instances being started or stopped the within the ConductR cluster.
+* All other endpoints is for configured using ConductR default configuration. The HTTP endpoint is exposed on the port `9443`
+
+### Deploying SSL certificate
+
+This step is optional.
+
+It's possible to deploy SSL cert as part of the configuration override. The SSL cert will enable HTTPS traffic which terminates at the HAProxy.
+
+Continuing from previous example, place the SSL certificate in `/tmp/custom-haproxy-conf`, e.g.
+
+```bash
+cp /tmp/test-cert.pem /tmp/custom-haproxy-conf
+```
+
+In the example above, the SSL cert called `test-cert.pem` is placed in `/tmp/custom-haproxy-conf`.
+
+The SSL cert will need to be referenced in the HAProxy configuration template. Continuing from previous example, uncomment the SSL configuration from the template, e.g.
+
+```
+...
+
+  {{#ifAcl 'public-site' '1' 'main'}}
+frontend my_www_frontend
+  bind {{haproxyHost}}:80
+  # Uncomment the following line if SSL is used
+  bind {{haproxyHost}}:443 ssl crt {{overrideConfigDir}}/test-cert.pem
+
+...
+
+```
+
+
+### Declare the custom HAProxy configuration template
+
+Create a file called `runtime-config.sh` within the proxy configuration directory, e.g.
+
+```bash
+touch /tmp/custom-haproxy-conf/runtime-config.sh
+```
+
+Populate the file with the following entry:
+
+```bash
+#!/bin/bash
+
+CONFIG_DIR=$( cd $( dirname "${BASH_SOURCE[0]}" ) && pwd )
+export CONDUCTR_HAPROXY_CONFIG_OVERRIDE="$CONFIG_DIR/haproxy-override.cfg"
+```
+
+The `CONFIG_DIR` refers to the directory where the configuration override will be expanded. `CONDUCTR_HAPROXY_CONFIG_OVERRIDE` is the environment variable which tells ConductR where to find the custom HAProxy configuration template.
+
+
+### Package the custom HAProxy configuration template
+
+Use the CLI to package the configuration override:
+
+```bash
+Felixs-MBP-2:~ felixsatyaputra$ shazar /tmp/custom-haproxy-conf
```

Comment:
Remove `Felixs-MBP-2:~ felixsatyaputra$`. In other bash examples we also do not indicate that this example needs to be executed in the terminal.